sklearn/metrics/classification.py

```diff
@@ -240,6 +240,8 @@ def confusion_matrix(y_true, y_pred, labels=None, sample_weight=None):
         labels = unique_labels(y_true, y_pred)
     else:
         labels = np.asarray(labels)
+        if np.all([l not in y_true for l in labels]):
```

Comment:
It can add up when there's no intersection and `labels` is large, but I think we're talking about a handful of milliseconds in most cases, so maybe it makes most sense to appeal to readability. I find the empty intersection version immediately clear, whereas the list comprehension takes a second to reason about.